resource_obmcs_core_internet_gateway.go

```diff
@@ -144,9 +144,14 @@ func (s *InternetGatewayResourceCrud) Get() (e error) {
 
 func (s *InternetGatewayResourceCrud) Update() (e error) {
 	opts := &baremetal.UpdateGatewayOptions{}
-	if isEnabled, ok := s.D.GetOk("enabled"); ok {
-		opts.IsEnabled = isEnabled.(bool)
-	}
+
+	// todo: GetOk malfunction with this bool: 'ok' is always the value of the bool
```

Comment:
Is this maybe https://github.com/hashicorp/terraform/issues/13512? We work around that in at least one other place.

Reply:
I don't think it's that, this has to do with how they verify zero values and do deep compare